### PR TITLE
feat: refactor pallet staking payouts

### PR DESCRIPTION
**Pull request by DylanVerstraete**

### CODE REVIEW

**Comment by DylanVerstraete** [comment]
https://github.com/threefoldtech/tfchain/issues/136

**Review by robvanmieghem — APPROVED** [review]
lgtm

File: pallet-staking/src/benchmarking.rs [hunk 1]
```diff
@@ -816,7 +820,7 @@ mod tests {
                     n,
                     <Test as Config>::MaxNominatorRewardedPerValidator::get() as u32,
                     false,
-                    RewardDestination::Staked,
+                    RewardDestination::Stash,
```
Comment (LeeSmet) [inline pallet-staking/src/benchmarking.rs]: Why did this get changed? it seems unrelated to the other changes

**Review by LeeSmet — CHANGES_REQUESTED** [review]

File: pallet-staking/src/benchmarking.rs (hunk 1 above)
Comment (DylanVerstraete) [reply pallet-staking/src/benchmarking.rs]: Unrelated change indeed

**Review by LeeSmet — APPROVED** [review]
